apps/extension/src/routes/popup/settings/settings-rpc.tsx

```diff
@@ -1,34 +1,54 @@
-import { AppQuerier } from '@penumbra-zone/query';
+import { QueryService } from '@buf/penumbra-zone_penumbra.connectrpc_es/penumbra/core/app/v1/app_connect';
+import { createGrpcWebTransport } from '@connectrpc/connect-web';
+import { createPromiseClient } from '@connectrpc/connect';
 import { FormEvent, useState } from 'react';
 import { Button, FadeTransition, Input } from '@penumbra-zone/ui';
-import { cn } from '@penumbra-zone/ui/lib/utils';
-import { useChainId } from '../../../hooks/chain-id';
+import { useChainIdQuery } from '../../../hooks/chain-id';
 import { ShareGradientIcon } from '../../../icons';
 import { SettingsHeader } from '../../../shared';
 import { useStore } from '../../../state';
 import { networkSelector } from '../../../state/network';
 import { internalSwClient } from '@penumbra-zone/router';
+import '@penumbra-zone/types/src/promise-with-resolvers';
 
 export const SettingsRPC = () => {
-  const { chainId, refetch } = useChainId();
+  const { chainId: currentChainId } = useChainIdQuery();
+  const [newChainId, setNewChainId] = useState(undefined as string | undefined);
   const { grpcEndpoint, setGRPCEndpoint } = useStore(networkSelector);
 
-  const [rpc, setRpc] = useState(grpcEndpoint ?? '');
-  const [rpcError, setRpcError] = useState(false);
+  const [rpcInput, setRpcInput] = useState(grpcEndpoint ?? '');
+  const [rpcError, setRpcError] = useState(undefined as string | undefined);
+  const [countdownTime, setCountdownTime] = useState(undefined as number | undefined);
 
-  const onSubmit = (e: FormEvent<HTMLFormElement>) => {
-    e.preventDefault();
-    void (async () => {
-      const querier = new AppQuerier({ grpcEndpoint: rpc });
+  const countdown = (seconds: number) => {
+    const { promise, resolve } = Promise.withResolvers<undefined>();
+    setCountdownTime(seconds);
+    setInterval(() => {
+      if (!seconds) resolve(undefined);
```

Comment:
surprised we need to pass `undefined` to resolve, thought values were optional here

Reply:
it would be optional if i used `Promise.withResolvers<void>` but eslint dislikes that

Reply:
updated the polyfill to default to a null parameter